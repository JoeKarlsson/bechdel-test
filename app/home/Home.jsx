/* jshint esversion: 6 */
'use strict';

import React from 'react';
import { Link } from 'react-router';
import NewFile from './NewFile.jsx';
import styles from './Home.scss';

export default React.createClass({
  getInitialState() {
    let filename = 'newFile'+ Math.floor(Math.random()*100) + '.txt';
    return {
      description: 'The description for this gist',
      files: [{
        filename: filename,
        originalFileName: filename,
        content: 'Add your content here'
      }]
    }
  },
  handleAddFile(e) {
    e.preventDefault();
    let newFile = {};
    let fileName = 'newFile'+ Math.floor(Math.random()*100) + '.txt';
    newFile.filename = fileName;
    newFile.originalFileName = fileName;
    newFile.content = 'Add your content here';
    this.state.files.push(newFile);
    this.setState({files: this.state.files});
  },
  handleDescriptionChange(e) {
    this.setState({ description: e.target.value });
  },
  handleFileNameChange(newFileName, originalFileName) {
    for (var i = this.state.files.length - 1; i >= 0; i--) {
      if (this.state.files[i].originalFileName === originalFileName) {
        this.state.files[i].filename = newFileName;
      }
    }
  },
  handleContentChange(content, filename) {
    for (var i = this.state.files.length - 1; i >= 0; i--) {
      if (this.state.files[i].filename === filename) {
        this.state.files[i].content = content;
      }
    }
  },
  handleFileDelete(filename) {
    let updatedFiles = this.state.files;
    for (var i = updatedFiles.length - 1; i >= 0; i--) {
      if (updatedFiles[i].filename === filename) {
        updatedFiles.splice(i, 1);
      }
    }
    this.setState({files: updatedFiles});
  },

  render() {
    const newFileNode = this.state.files.map(function(fileData) {
      return (
        <NewFile
          file={fileData}
          onFileNameChange={this.handleFileNameChange}
          onContentChange={this.handleContentChange}
          onFileDelete={this.handleFileDelete}
          key={fileData.filename + Math.random()*100000}
        />
      )
    }.bind(this));
    return (
      <div>
        <div className="section header">
          <div className="container">
            <h2 className="section-heading">Instantly share code, notes, and snippets.</h2>
            <Link to={'/login'}><button className="button button-primary">Let's get started</button></Link>
          </div>
        </div>

        <div className="new-btns">
          <div className="cancel-add-file-btns">
            <button onClick={this.handleAddFile}>Add File</button>
          </div>
          <div className="create-public-private-btns">
            <Link to={'/login'}><button>Create Public Gist</button></Link>
            <Link to={'/login'}><button>Create Private Gist</button></Link>
          </div>
        </div>
        <form>
          <label for="description">Description</label>
          <input
            ref="description"
            type='text'
            id='description'
            className="u-full-width"
            placeholder={this.state.description}
            value={this.state.description}
            defaultValue={this.state.description}
            onChange={this.handleDescriptionChange}
          />

          { newFileNode }

          <div className="new-btns">
            <div className="cancel-add-file-btns">
              <button onClick={this.handleAddFile}>Add File</button>
            </div>
            <div className="create-public-private-btns">
              <Link to={'/login'}><button>Create Public Gist</button></Link>
              <Link to={'/login'}><button>Create Private Gist</button></Link>
            </div>
          </div>
        </form>
      </div>
    )
  }
});