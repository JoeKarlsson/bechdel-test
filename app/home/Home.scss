.section {
  padding: 8rem 0 7rem;
  text-align: center;
}
.section-heading,
.section-description {
  margin-bottom: 1.2rem;
}