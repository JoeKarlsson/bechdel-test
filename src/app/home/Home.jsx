import React from 'react';
import styles from './Home.scss';

class Home extends React.Component {
  render() {
    return (
      <div className={styles.home}>Home</div>
    );
  }
}

export default Home;
