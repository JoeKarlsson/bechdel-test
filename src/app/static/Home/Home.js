import React from 'react';
import Hero from './Hero/Hero';
import Films from '../../films/Films/FilmsContainer';

const Home = () => {
	return (
		<div className="Home">
			<Hero />
			<Films />
		</div>
	);
};

export default Home;
