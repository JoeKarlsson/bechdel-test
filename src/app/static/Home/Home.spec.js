import React from 'react';
import Home from './Home';

describe('Home Page', () => {
	describe('rendering', () => {
		it('match the snapshot', () => {
			const wrapper = shallow(<Home />);
			expect(wrapper).toMatchSnapshot();
		});
	});
});
