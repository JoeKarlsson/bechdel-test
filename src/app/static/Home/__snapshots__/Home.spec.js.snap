// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page rendering match the snapshot 1`] = `ShallowWrapper {}`;
